src/transformers/keras_callbacks.py

```diff
@@ -193,8 +193,7 @@ def on_epoch_end(self, epoch, logs=None):
             # This dense conditional recognizes the case where we have an encoder-decoder model, but
             # avoids getting tangled up when we just have a model with a layer called 'encoder'
             if hasattr(self.model, "encoder") and hasattr(self.model.encoder, "main_input_name"):
-                if self.model.encoder.main_input_name != self.model.main_input_name:
-                    main_input_name = self.model.encoder.main_input_name
+                main_input_name = self.model.encoder.main_input_name
```

Comment:
we can set to this value no matter if `self.model.encoder.main_input_name != self.model.main_input_name` is True or False.